services/src/handlers/datasets.rs

```diff
@@ -1404,83 +1404,83 @@ mod tests {
         .map_err(Into::into)
     }
 
-    #[tokio::test(flavor = "multi_thread", worker_threads = 1)]
-    async fn create_dataset() -> Result<()> {
-        let mut test_data = TestDataUploads::default(); // remember created folder and remove them on drop
+    fn my_chunk_byte_size() -> geoengine_operators::engine::ChunkByteSize {
+        TestDefault::test_default()
+    }
 
-        let exe_ctx_tiling_spec = TilingSpecification {
+    fn my_exe_ctx_tiling_spec() -> TilingSpecification {
+        TilingSpecification {
             origin_coordinate: (0., 0.).into(),
             tile_size_in_pixels: GridShape2D::new([600, 600]),
-        };
+        }
+    }
 
-        // override the pixel size since this test was designed for 600 x 600 pixel tiles
-        with_temp_context_from_spec(
-            exe_ctx_tiling_spec,
-            TestDefault::test_default(),
-            |app_ctx, _| async move {
-                let session_id = app_ctx.default_session_id().await;
+    #[getest::test(
```

Comment:
I am not sure, but I think the semantics of the test changed here. Previously the test returned a `Result` which means at the end of the test it was checked if the body returned `Ok`. I f I now look at the code created by the macro, it no longer return (or unwraps) the error.

Reply:
what I mean is: the `with_temp_context_from_spec` return a `Result` which you ignore. 

When I let rust-analyzer expand the macro it generates:

```
...
 crate::util::tests::with_temp_context_from_spec(
            tiling_spec,
            query_ctx_chunk_size,
            create_dataset,
        )
        .await;

    ...
```

I think you have to unwrap here.

Reply:
No, you are wrong here. `with_…_spec` returns `R`, which means the return type of `create_dataset`. This can be a `Result`, but mostly it is void.

Nevertheless, I changed the macro in a way that you can have both tests that return void and tests that return a `Result`.

[11c579d](https://github.com/geo-engine/geoengine/pull/859/commits/11c579dfca6287afbe4020183caca5f75bedd3c5)